core/CronArchive/SegmentArchivingRequestUrlProvider.php

```diff
@@ -96,12 +95,31 @@ public function getUrlParameterDateString($idSite, $period, $date, $segment)
         return $date;
     }
 
-    private function getOldestDateToProcessForNewSegment(Date $segmentCreatedTime)
+    private function getOldestDateToProcessForNewSegment($idSite, $segment)
     {
+        /**
+         * @var Date $segmentCreatedTime
+         * @var Date $segmentLastEditedTime
+         */
+        list($segmentCreatedTime, $segmentLastEditedTime) = $this->getCreatedTimeOfSegment($idSite, $segment);
+
         if ($this->processNewSegmentsFrom == self::CREATION_TIME) {
             $this->logger->debug("process_new_segments_from set to segment_creation_time, oldest date to process is {time}", array('time' => $segmentCreatedTime));
 
             return $segmentCreatedTime;
+        } elseif ($this->processNewSegmentsFrom == self::LAST_EDIT_TIME) {
+            $this->logger->debug("process_new_segments_from set to segment_last_edit_time, segment last edit time is {time}",
+                array('time' => $segmentLastEditedTime));
+
+            if ($segmentLastEditedTime === null
```

Comment:
I actually thought we did and was just being extra careful (didn't want to slow core:archive down if somehow the field was null (maybe due to a plugin)), but it turns out we don't initialize it to the created time... more strangeness in the world of Piwik core.

I don't think it's necessary to change this for 2.15, but it seems obvious to me that the last edit time should be initialized to the created time. If you agree, I'll create an issue for 3.0.